frontend/src/redux/note-details/raw-note-frontmatter-parser/parser.ts

```diff
@@ -51,6 +52,40 @@ const parseRawNoteFrontmatter = (rawData: RawNoteFrontmatter): NoteFrontmatter =
   }
 }
 
+/**
+ * Schema for the {@link RawNoteFrontmatter} for runtime validation.
+ */
+const RawNoteFrontmatterSchema = Joi.object({
+  title: Joi.string().optional().allow(null),
+  description: Joi.string().optional(),
+  tags: Joi.alternatives(Joi.string(), Joi.number(), Joi.array().items(Joi.string())).optional(),
+  robots: Joi.string().optional(),
+  lang: Joi.string().optional(),
+  dir: Joi.string().optional(),
+  breaks: Joi.boolean().optional(),
+  GA: Joi.string().optional(),
+  disqus: Joi.string().optional(),
+  license: Joi.string().optional(),
+  type: Joi.string().optional(),
+  slideOptions: Joi.object().allow(null).pattern(/.*/, Joi.string().allow('')),
+  opengraph: Joi.object().allow(null).pattern(/.*/, Joi.string().allow(''))
```

Comment:
Why should objects validate against a regex pattern?
You could verify their structure as well. 
There is also a type for SlideOptions in our code. 

Is there a reason why you're allowing null here instead of optional() ?

Reply:
Hi, sorry for no response. 
I am not sure but there is one test that checks that title is null (when setted by default). And optional checks only if field undefined. 

So if you provide null to title and check it with joy => then error

Reply:
which test?

Reply:
anyway. slideOptions and opengraph shouldnt be empty. They should be optional.  I don't know why the rawfrontmatter type accepts null. It should be optional. Maybe we need to change the types a bit. But not in this PR. anyway. Empty string shouldn't be a valid value for opengraph or slideoptions because they're optional objects.